### PR TITLE
chore(docs): Improves the pick quasar flavor page

### PR DESCRIPTION
**Pull request by smolinari**
There was some confusion from a user about what is meant with the build paths (mobile and electron being on one line), so I separated them, and I also decided to improve the page somewhat (and from my perspective).

Scott

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: docs/src/pages/start/pick-quasar-flavour.md [hunk 1]
```diff
@@ -21,22 +19,31 @@ Here's a quick comparison:
...
+| Build SSR (+ optional PWA client takeover)                                                 | -       | **Yes**    | Yes(*)                          |
+| Build Mobile Apps via Cordova or Capacitor                                                 | **Yes** | **Yes**    | Yes(*)                          |
+| Develop Mobile Apps with HMR directly on your phone.                                       | -       | **Yes**    | Yes(*) |
+| Build Desktop Apps via Electron                                                            | -       | **Yes**    | Yes(*)                             |
+| Build Browser Extensions                                                                   | -       | **Yes**    | Yes(*)                          |
 | Quasar **App Extensions**                                                                  | -       | **Yes**    | - |
-| App icons & splash screens easy management through [Icon Genie](/icongenie/introduction) | - | **Yes** | - |
+| Easy management of App icons & splash screens via [Icon Genie](/icongenie/introduction)    | - | **Yes** | - |
```
Comment (hawkeye64) [inline docs/src/pages/start/pick-quasar-flavour.md:28]: I suggest making this "Icon Genie Cli". Too many people getting mixed up with older AE version.

File: docs/src/pages/start/pick-quasar-flavour.md [hunk 2]
```diff
@@ -21,22 +19,31 @@ Here's a quick comparison:
...
 | Dynamic RTL support for Quasar components                                                  | **Yes** | **Yes**    | **Yes**                       |
 | Generating your own website/app RTL equivalent CSS rules automatically by Quasar           | -       | **Yes**    | **Yes**                       |
 | **Ensure everything "simply works" out of the box**, using latest and greatest Quasar specs.   | -       | **Yes**    | -                             |
 | **Tight integration between build modes**, taking full advantage of all Quasar's capabilities. | -       | **Yes**    | -                             |
-| Develop mobile apps with HMR directly on your phone.                                       | -       | **Yes**    | Yes*                          |
```
Comment (hawkeye64) [inline docs/src/pages/start/pick-quasar-flavour.md:33]: This is an important feature. Why remove it?

File: docs/src/pages/start/pick-quasar-flavour.md [hunk 3]
```diff
@@ -21,22 +19,31 @@ Here's a quick comparison:
...
-| One codebase to rule SPA, PWA, SSR, Mobile Apps, Electron Apps                             | -       | **Yes**    | Yes*                          |
+| One codebase to create SPA, PWA, SSR, Mobile Apps, Electron Apps                             | -       | **Yes**    | Yes(*)                      |
 | Tree Shaking                                                                               | -       | **Yes**    | **Yes**                       |
 | SFC (Single File Component - for Vue) support                                              | -       | **Yes**    | **Yes**                       |
 | Advanced configuration through dynamic quasar.conf.js                                      | -       | **Yes**    | -                             |
 | Unit & end to end testing                                                                  | -       | **Yes**    | **Yes**                       |
+| TypeScript support                                                              | -       | **Yes**    | **Yes**                       |
+|**Best and Most Popular Choice!**  |  |**Quasar CLI** |  |
```
Comment (hawkeye64) [inline docs/src/pages/start/pick-quasar-flavour.md:39]: This looks confusing to me. All the rest have **Yes**

File: docs/src/pages/start/pick-quasar-flavour.md [hunk 4]
```diff
@@ -21,22 +19,31 @@ Here's a quick comparison:
...
 
-::: tip Footnote for above
-\* Cross-platform support with Vue CLI is handled by community plugins. These are not tightly integrated with Quasar as with Quasar CLI and so they may have issues. Also, Quasar CLI ensures Quasar is used at its best (including performance-wise).
+
+::: tip (*)Important!
+Although you may get a similar multi-platform support via the Vue CLI and some Vue community built plugins, these 3rd party supported build paths aren't tightly integrated with Quasar's components. Thus, as you run into problems with these 3rd party plugins, you will have to depend on the support of each individual plugin developer. With Quasar, you have a one-stop-shop should anything go wrong. Also, the Quasar CLI ensures applications are built to the best possible standards in both performance, project size and best practices. You will find no such guarantees anywhere else!
```
Comment (hawkeye64) [inline docs/src/pages/start/pick-quasar-flavour.md:43]: Feels slightly off. Maybe:
"You may get a similar multi-platform support via the Vue CLI and some Vue community built plugins, but these 3rd party supported build paths aren't tightly integrated with Quasar's components."

**Review by hawkeye64 — CHANGES_REQUESTED** [review]

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 1 above)
Comment (webnoob) [reply docs/src/pages/start/pick-quasar-flavour.md:28]: I agree but with CLI in caps :)

File: docs/src/pages/start/pick-quasar-flavour.md [hunk 5]
```diff
@@ -21,22 +19,31 @@ Here's a quick comparison:
...
+| Easy management of App icons & splash screens via [Icon Genie](/icongenie/introduction)    | - | **Yes** | - |
 | Dynamic RTL support for Quasar components                                                  | **Yes** | **Yes**    | **Yes**                       |
 | Generating your own website/app RTL equivalent CSS rules automatically by Quasar           | -       | **Yes**    | **Yes**                       |
 | **Ensure everything "simply works" out of the box**, using latest and greatest Quasar specs.   | -       | **Yes**    | -                             |
 | **Tight integration between build modes**, taking full advantage of all Quasar's capabilities. | -       | **Yes**    | -                             |
-| Develop mobile apps with HMR directly on your phone.                                       | -       | **Yes**    | Yes*                          |
-| One codebase to rule SPA, PWA, SSR, Mobile Apps, Electron Apps                             | -       | **Yes**    | Yes*                          |
+| One codebase to create SPA, PWA, SSR, Mobile Apps, Electron Apps                             | -       | **Yes**    | Yes(*)                      |
```
Comment (webnoob) [inline docs/src/pages/start/pick-quasar-flavour.md:33]: Don't forget BEX 😋

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 2 above)
Comment (webnoob) [reply docs/src/pages/start/pick-quasar-flavour.md:33]: @hawkeye64 Looks like it's covered by the new addition of the one codebase.

@smolinari I think it's worth *keeping* this line (so it's very clear about mobile apps) but also keeping the new one you've added. The reason being, this line covers HMR *on* the device which isn't a given in all frameworks.

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 3 above)
Comment (webnoob) [reply docs/src/pages/start/pick-quasar-flavour.md:39]: Agreed. This should read Yes.

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 1 above)
Comment (smolinari) [reply docs/src/pages/start/pick-quasar-flavour.md:28]: Done.

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 2 above)
Comment (smolinari) [reply docs/src/pages/start/pick-quasar-flavour.md:33]: I didn't. I moved it up under the "mobile apps" line.

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 3 above)
Comment (smolinari) [reply docs/src/pages/start/pick-quasar-flavour.md:39]: Changed.

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 4 above)
Comment (smolinari) [reply docs/src/pages/start/pick-quasar-flavour.md:43]: The sentence is grammatically correct. The "Although" is like a "but", just put at the front of the sentence to show from the start, the path of using the Quasar Plugin with Vue CLI is questionable. 

https://www.writewithjean.com/usage-and-grammar/although/

Scott

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 5 above)
Comment (smolinari) [reply docs/src/pages/start/pick-quasar-flavour.md:33]: BEX is in there. :)

**Comment by smolinari** [comment]
Review suggestions committed. 

Scott

**Comment by smolinari** [comment]
The Mobile App HMR bit is still there, just higher up on the list. 

Scott

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 5 above)
Comment (webnoob) [reply docs/src/pages/start/pick-quasar-flavour.md:33]: `One codebase to create SPA, PWA, SSR, Mobile Apps, Electron Apps` ? Am I losing my mind?

Comment (webnoob) [reply docs/src/pages/start/pick-quasar-flavour.md:33]: I'm thinking `One codebase to create SPA, PWA, SSR, Mobile Apps, Electron Apps, Browser Extensions`

Comment (smolinari) [reply docs/src/pages/start/pick-quasar-flavour.md]: Got it. Updated.

File: docs/src/pages/start/pick-quasar-flavour.md (hunk 4 above)
Comment (webnoob) [reply docs/src/pages/start/pick-quasar-flavour.md:43]: I trust Scotts grammar so resolving this so we can merge :)

File: docs/src/pages/start/pick-quasar-flavour.md [hunk 6]
```diff
@@ -21,22 +19,31 @@ Here's a quick comparison:
...
-| Develop mobile apps with HMR directly on your phone.                                       | -       | **Yes**    | Yes*                          |
-| One codebase to rule SPA, PWA, SSR, Mobile Apps, Electron Apps                             | -       | **Yes**    | Yes*                          |
+| One codebase to create SPA, PWA, SSR, Mobile Apps, Electron Apps and Browser Extensions                             | -       | **Yes**    | Yes(*)                      |
 | Tree Shaking                                                                               | -       | **Yes**    | **Yes**                       |
 | SFC (Single File Component - for Vue) support                                              | -       | **Yes**    | **Yes**                       |
 | Advanced configuration through dynamic quasar.conf.js                                      | -       | **Yes**    | -                             |
 | Unit & end to end testing                                                                  | -       | **Yes**    | **Yes**                       |
```
Comment (IlCallo) [inline docs/src/pages/start/pick-quasar-flavour.md:37]: This should probably be renamed to "Unit & end to end testing integration" or "Unit & end to end testing support", as Quasar isn't automatically unit or e2e tested per se

Comment (smolinari) [reply docs/src/pages/start/pick-quasar-flavour.md]: I think it's fairly clear the project target is meant, not the core software. :) But, I'll make the change, so it's definitely not mistaken.

**Pull request merged**